src/Pages/Dashboard/Sensor.jsx

```diff
@@ -1,62 +1,118 @@
-import React, { PureComponent } from "react";
+import React, {PureComponent} from "react";
 // Material UI Components
-import { withStyles } from "@material-ui/core";
+import {withStyles} from "@material-ui/core";
 //Style
-import { dashboardStyle } from "./dashboardStyle";
+import {dashboardStyle} from "./dashboardStyle";
 import Dashboard from "./Dashboard";
 
 class Sensor extends PureComponent {
-  constructor(props) {
-    super(props);
-    this.db = null;
-    this.state = {
-      data: []
+    constructor(props) {
+        super(props);
+        this.db = null;
+        this.state = {
+            data: [],
+            historydata: []
+        };
+        let date = new Date();
+        let now = date.getTime();
+        let d = now - (30 * 24 * 60 * 60 * 1000);
+        this.timestamp = Math.floor(d / 1000);
+        this.nowstamp = Math.floor(now / 1000);
+    }
+
+    // We wait until the component has been mounted (which means
+    // other components are unmounted) before polling, so we don't
+    // have multiple registers to a particular database
+    componentDidMount = () => {
+        this.getDatabase(this.props.sensorId);
+    };
+
+    changeRange = (value) => {
+        this.setState({selected: value});
+        let offset;
+        if (value === 1) {
+            offset = 60 * 60 * 1000;
+        } else if (value === 2) {
+            offset = 2 * 60 * 60 * 1000;
+        } else if (value === 3) {
+            offset = 6 * 60 * 60 * 1000;
+        } else if (value === 4) {
+            offset = 24 * 60 * 60 * 1000;
+        } else if (value === 5) {
+            offset = 7 * 24 * 60 * 60 * 1000;
+        } else if (value === 6) {
+            offset = 30 * 24 * 60 * 60 * 1000;
+        } else if (value === 7) {
+            offset = 183 * 24 * 60 * 60 * 1000;
+        } else if (value === 8) {
+            offset = 365 * 24 * 60 * 60 * 1000;
+        } else {
+            offset = 6 * 60 * 60 * 1000;
+        }
+        let date = new Date();
+        let now = date.getTime();
+        let d = now - offset;
+        this.timestamp = Math.floor(d / 1000);
+        this.nowstamp = Math.floor(now / 1000);
+        this.setState({range: this.timestamp});
+        this.getDatabase(this.props.sensorId);
```

Comment:
Have you confirmed whether the case of switching to the history tab then back to the data tab won't affect whether or not we lose the database references? `this.getDatabase()` will give us a new db reference for the history tab, overwriting the original one (which can cause a memory leak). 

I apologize as I haven't been available to help you out with this one much. I can show you how to run the scripts to push arbitrary data so that you can test it while pulling live data.

Reply:
I might need some help with this one... I will try to fix others first.

Reply:
I tried the send random data script, and it doesn't seem to cause any errors. The code did not show what I expected exactly (show 20 days history feature would not limit to 30 entries) so I fixed that. I don't see any further problems than that. Can you take a look at my new commits?

Reply:
OK, I do see that now that one of my sensors has data on it (that the main page has something to show), when I change to the history page, the graph won't be empty but instead shows a couple of readings.

Reply:
> I tried the send random data script

Is this the one that's on your branch? It's stale and it _shouldn't_ work anymore. If it does, then I screwed the pooch on the database permissions. Can you confirm that you were in fact able to push data into the database?